x-pack/test/functional/services/index.ts

```diff
@@ -88,5 +101,15 @@ export const services = {
   uptime: UptimeProvider,
   infraSourceConfigurationForm: InfraSourceConfigurationFormProvider,
   infraLogStream: InfraLogStreamProvider,
-  ml: MachineLearningProvider,
+  mlAnomalyExplorer: MachineLearningAnomalyExplorerProvider,
```

Comment:
I think we could probably flatten these services a little, maybe in a follow up PR. Consider an `ml` service that looks something like this:

```js
import {
  MachineLearningAnomalyExplorerProvider,
  MachineLearningAPIProvider,
  MachineLearningDataFramesProvider,
  MachineLearningDataVisualizerProvider,
  MachineLearningJobManagementProvider,
  MachineLearningJobSourceSelectionProvider,
  MachineLearningJobTypeSelectionProvider,
  MachineLearningJobWizardCommonProvider,
  MachineLearningNavigationProvider,
  MachineLearningSettingsProvider,
  MachineLearningSingleMetricViewerProvider,
} from './machine_learning';

export function MachineLearningProvider(context: FtrProviderContext) {
  const anomalyExplorer = MachineLearningAnomalyExplorerProvider(context);
  const api = MachineLearningAPIProvider(context);
  const dataFrames = MachineLearningDataFramesProvider(context);
  const dataVisualizer = MachineLearningDataVisualizerProvider(context);
  const jobManagement = MachineLearningJobManagementProvider(context);
  const jobSourceSelection = MachineLearningJobSourceSelectionProvider(context);
  const jobTypeSelection = MachineLearningJobTypeSelectionProvider(context);
  const jobWizardCommon = MachineLearningJobWizardCommonProvider(context);
  const navigation = MachineLearningNavigationProvider(context);
  const settings = MachineLearningSettingsProvider(context);
  const singleMetricViewer = MachineLearningSingleMetricViewerProvider(context);

  return {
    anomalyExplorer,
    api,
    dataFrames,
    dataVisualizer,
    jobManagement,
    jobSourceSelection,
    jobTypeSelection,
    jobWizardCommon,
    navigation,
    settings,
    singleMetricViewer,
  }
}
```

Then you can get the `ml` service and use things like `ml.api` or `ml.jobManagement` and it will all be properly types with TypeScript too!

Reply:
Great idea, thanks! Done.